chapters/classes.tex

```diff
@@ -601,6 +601,11 @@ \subsection{Discrete-Time Variables}\label{discrete-time-variables}
 A \lstinline!Real! variable assigned in a \lstinline!when!-clause is a discrete-time variable, even though it was not declared with the prefix \lstinline!discrete!.
 A \lstinline!Real! variable not assigned in any \lstinline!when!-clause and without any type prefix is a continuous-time variable.
 
+The determination of whether a variable is assigned in \lstinline!when!-clause requires evaluating the controlling conditions of any enclosing \lstinline!if!-equations and \lstinline!for!-equations, \cref{where-a-when-equation-may-occur}.
+\begin{nonnormative}
+Checking whether a \lstinline!Real! variable declared as \lstinline!discrete! is assigned in a \lstinline!when!-clause for other valid values of controlling conditions is a quality of implementation issue for tools.
```

Comment:
I don't think we every say _… for tools_ when talking out quality of implementation:
```suggestion
It is a quality of implementation to check that a variable declared as \lstinline!discrete! is assigned in a \lstinline!when!-clause also for other valid values of controlling conditions.
```